api/src/test/java/org/openmrs/api/ConceptServiceTest.java

```diff
@@ -3531,4 +3536,76 @@ public void getConcepts_shouldPassWithAndOrNotWords() {
 		Assert.assertEquals(1, searchResults.size());
 		assertThat(searchResults.get(0).getWord(), is("SALBUTAMOL INHALER NOT"));
 	}
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldFindAConceptByItsConceptId() {
+		String id = "5089";
+		Concept conceptById = Context.getConceptService().getConceptByNameOrIdOrMap(id);
+		assertEquals("WEIGHT (KG)", conceptById.getName().toString());
+	}
+
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldGetConceptByName() {
+		String nameToFetch = "WEIGHT (KG)";
+		Concept concept = Context.getConceptService().getConceptByNameOrIdOrMap(nameToFetch);
+		Assert.assertEquals("5089", concept.getConceptId().toString());
+	}
+
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldFindAConceptByItsMapping() {
+		String mapping = "SSTRM:454545";
+		Concept concept = conceptService.getConceptByNameOrIdOrMap(mapping);
+		Assert.assertEquals(24, concept.getId().intValue());
+	}
+
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldFindAConceptByItsUuid() {
+			String uuid = "0cbe2ed3-cd5f-4f46-9459-26127c9265ab";
+			Concept concept = Context.getConceptService().getConceptByNameOrIdOrMap(uuid);
+			Assert.assertEquals(3, (int) concept.getConceptId());
+	}
+
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldNotFindAConceptWithInvalidUuid() {
+		String invalidUuid = "1000";
+		executeDataSet(INITIAL_CONCEPTS_XML);
+		Assert.assertNull(conceptService.getConceptByNameOrIdOrMap(invalidUuid));
+	}
+
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldFindAConceptByStaticConstant() {
+		assertThat(conceptService.getConceptByNameOrIdOrMap("org.openmrs.api.ConceptServiceTest.TEST_CONCEPT_CONSTANT_ID"), notNullValue());
+		assertThat(conceptService.getConceptByNameOrIdOrMap("org.openmrs.api.ConceptServiceTest.TEST_CONCEPT_CONSTANT_UUID"), notNullValue());
+		assertThat(conceptService.getConceptByNameOrIdOrMap("org.openmrs.api.ConceptServiceTest.TEST_CONCEPT_CONSTANT_NAME"), notNullValue());
+	}
+
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldReturnNullOtherwise() {
```

Comment:
I think you should name this test method like `getConceptByNameOrIdOrMap_shouldReturnNullWhen< write condtition here like ConceptRefDoesNotExist >` 

NullOtherwise is not informative.